compose/material3/material3/src/jsWasmMain/kotlin/androidx/compose/material3/PlatformDateFormat.jsWasm.kt

```diff
@@ -0,0 +1,207 @@
+        utcTimeMillis: Long,
+        skeleton: String,
+        locale: CalendarLocale
+    ): String {
+        val date = Date(utcTimeMillis)
+
+        return date.toLocaleDateString(
+            locales = locale.toLanguageTag(),
+            options = dateLocaleOptions {
+                year = when {
+                    skeleton.contains("y", true) -> NUMERIC
+                    else -> undefined
+                }
+                month = when {
+                    skeleton.contains("MMMM", true) -> LONG
+                    skeleton.contains("MMM", true) -> SHORT
+                    skeleton.contains("M", true) -> NUMERIC
+                    else -> undefined
+                }
+                day = when {
+                    skeleton.contains("d", true) -> NUMERIC
+                    else -> undefined
+                }
+                hour = when {
+                    skeleton.contains("h", true) -> NUMERIC
+                    else -> undefined
+                }
+                minute = when {
+                    skeleton.contains("i", true) -> NUMERIC
+                    else -> undefined
+                }
+                second = when {
+                    skeleton.contains("s", true) -> NUMERIC
+                    else -> undefined
+                }
+            }
+        )
+    }
+
+    actual fun parse(
+        date: String,
+        pattern: String
+    ): CalendarDate? {
+        val year = parseSegment(date, pattern, "yyyy") ?: return null
+        val month = parseSegment(date, pattern, "mm") ?: return null
+        val day = parseSegment(date, pattern, "dd")
+
+        return LocalDate(
+            year, month, day ?: 1
+        ).atStartOfDayIn(TimeZone.currentSystemDefault())
+            .toCalendarDate()
+    }
+
+    private fun parseSegment(date: String, pattern: String, segmentPattern: String): Int? {
+        val index = pattern
+            .indexOf(segmentPattern, ignoreCase = true)
+            .takeIf { it >= 0 } ?: return null
+
+        return date.substring(index, index + segmentPattern.length)
+            .toIntOrNull()
+    }
+
+    actual fun getDateInputFormat(locale: CalendarLocale): DateInputFormat {
+        val date = Date(year = 1234, month = 10, day = 23)
+
+        val shortDate = date.toLocaleDateString(locale.toLanguageTag())
+
+        val delimiter = shortDate.first { !it.isDigit() }
+
+        val pattern = shortDate
+            .replace("1234", "yyyy")
+            .replace("11", "MM") //10 -> 11 not an error. month is index
+            .replace("23", "dd")
+
+        return DateInputFormat(pattern, delimiter)
+    }
+
+    private fun weekdayNames(): List<Pair<String, String>> {
+        val now = Date.now()
+
+        val week = List(DaysInWeek) {
+            Date(now + MillisecondsIn24Hours * it)
+        }.sortedBy { it.getDay() } // sunday to saturday
+
+        val mondayToSunday = week.drop(1) + week.first()
+
+        val locale = defaultLocale()
```

Comment:
Could `Locale.current` be used here instead?

Reply:
Sure. defaultLocaleis resolves to Locale.current. At first I wanted to use this snippet everywhere to commonize receiving current locale. Is it a good intention or is it better to use platform way directly to do it?